Implement more counters modes.

qnikst opened the pull request:
New counters were implemented:
  - bytes - calculate amount of bytes allocated by each function.
  - ticks - calculate number of ticks spent in each function.

Main rationale is that alloc and time in percentage is fairily
unusable in non-trivial programs or programs that spends much time
in idle. So both bytes and ticks gives better stats.

Additional modes require program to be run with -P, so additional
checks were introduced in order to notify user about this requirement.

bitonic commented:
Thanks, I'll try to get to this ASAP (hopefully before the end of the week).

bitonic commented on ghc-prof-flamegraph.cabal:
```diff
@@ -6,6 +6,7 @@ license-file:        LICENSE
 author:              Francesco Mazzoli
 maintainer:          francesco@fpcomplete.com
 copyright:           (c) 2015 FP Complete Corporation
+                     (c) 2016 Seagate Technology Limited
```
I'm not sure this copyright notice is warranted, how does it usually work in these cases? In my experience, PRs do not result in a new copyright notice in the .cabal file.

bitonic commented on ghc-prof-flamegraph.hs:
```diff
@@ -9,35 +9,64 @@ import qualified Options.Applicative as Opts
 optionsParser = Options
-  <$> Opts.switch (Opts.long "alloc" <> Opts.help "Uses the allocation measurements instead of time measurements")
+  <$> (Opts.flag' Alloc (Opts.long "alloc" <> Opts.help "Uses the allocation measurements instead of time measurements")
+       Opts.<|> Opts.flag' Entries (Opts.long "entry" <> Opts.help "Usues entries the measurements instead of time measurements")
```
Typo: `Usues` => `Uses`

bitonic replied:
Minor comment: I'd use the unqualified `<|>` from `Control.Applicative`. Not a blocker.

bitonic commented on ghc-prof-flamegraph.hs:
```diff
@@ -46,4 +75,7 @@ main = do
-    Right ls -> putStr $ unlines $ generateFrames options ls
+    Right (names, ls) ->
+      case checkNames (optionsReportType options) names of
+        Just problem -> putStrLn problem
```
Please use `error` or exit with exit code 1, since this signals a problem.

bitonic commented on ghc-prof-flamegraph.hs:
```diff
@@ -9,35 +9,64 @@ import qualified Options.Applicative as Opts
+  | otherwise        = Just "No ticks information, please run program with +RTS -P"
+
+normalize :: ReportType -> Double -> Int
+normalize Alloc = round . (10 *)
```
Could you leave a comment in the code on why this measure is needed?

qnikst replied:
I kept this logic from existing code, as far as I understood idea was to find portion of the 'unknown'. I'd happily remove that part if I knew it not obvious or needed.

bitonic replied:
Aha, sorry, I missed that! Let's leave it as-is, I do not recall right now why this was needed. I should have added a comment then!

qnikst commented:
Thanks for the review, I'll address all of the comments today.

qnikst commented:
I've addressed all the comments. Additionally I've added `--time` flag and make it default, this flag return old behavior. Also I've removed copyright notice, because I'm not entirely sure about all details of  the procedure.

bitonic commented:
Thanks!
